src/logics/ArithLogic.h

```diff
@@ -0,0 +1,473 @@
+    static const std::string tk_int_zero;
+    static const std::string tk_real_one;
+    static const std::string tk_int_one;
+    static const std::string tk_real_neg;
+    static const std::string tk_int_neg;
+    static const std::string tk_real_minus;
+    static const std::string tk_int_minus;
+    static const std::string tk_real_plus;
+    static const std::string tk_int_plus;
+    static const std::string tk_real_times;
+    static const std::string tk_int_times;
+    static const std::string tk_real_div;
+    static const std::string tk_int_div;
+    static const std::string tk_int_mod;
+    static const std::string tk_real_leq;
+    static const std::string tk_int_leq;
+    static const std::string tk_real_lt;
+    static const std::string tk_int_lt;
+    static const std::string tk_real_geq;
+    static const std::string tk_int_geq;
+    static const std::string tk_real_gt;
+    static const std::string tk_int_gt;
+    static const std::string s_sort_real;
+    static const std::string s_sort_int;
+
+    // Reals
+    SRef                sort_REAL;
+    PTRef               term_Real_ZERO;
+    PTRef               term_Real_ONE;
+    PTRef               term_Real_MINUSONE;
+
+    SymRef              sym_Real_ZERO;
+    SymRef              sym_Real_ONE;
+    SymRef              sym_Real_NEG;
+    SymRef              sym_Real_MINUS;
+    SymRef              sym_Real_PLUS;
+    SymRef              sym_Real_TIMES;
+    SymRef              sym_Real_DIV;
+    SymRef              sym_Real_EQ;
+    SymRef              sym_Real_LEQ;
+    SymRef              sym_Real_LT;
+    SymRef              sym_Real_GEQ;
+    SymRef              sym_Real_GT;
+    SymRef              sym_Real_ITE;
+    SymRef              sym_Real_DISTINCT;
+
+
+    // Integers
+    SRef                sort_INT;
+    PTRef               term_Int_ZERO;
+    PTRef               term_Int_ONE;
+    PTRef               term_Int_MINUSONE;
+    SymRef              sym_Int_ZERO;
+    SymRef              sym_Int_ONE;
+    SymRef              sym_Int_NEG;
+    SymRef              sym_Int_MINUS;
+    SymRef              sym_Int_PLUS;
+    SymRef              sym_Int_TIMES;
+    SymRef              sym_Int_DIV;
+    SymRef              sym_Int_MOD;
+    SymRef              sym_Int_EQ;
+    SymRef              sym_Int_LEQ;
+    SymRef              sym_Int_LT;
+    SymRef              sym_Int_GEQ;
+    SymRef              sym_Int_GT;
+    SymRef              sym_Int_ITE;
+    SymRef              sym_Int_DISTINCT;
+
+public:
+    enum class ArithType {
+        LIA, LRA, NIA, NRA, LIRA, NIRA, RDL, IDL
+    };
+
+private:
+    ArithType           arithType;
+    static std::vector<std::string> const logicNames;
+    static vec<opensmt::Logic_t> const logicTypes;
+
+public:
+    ArithLogic(ArithType arithType);
+    ~ArithLogic() { for (auto number : numbers) { delete number; } }
+    const char       *getName() const override { return logicNames[static_cast<int>(arithType)].data(); }
```

Comment:
Could this return `std::string const &` instead?

Reply:
I can turn it to `std::string`.  It doesn't seem right to turn it to a `std::string const &` right now since the logic currently doesn't really know its name statically.  With the new idea discussed today about passing a structure of booleans in the constructor we could pass the name as well (as an `std::string`) and then have a single implementation of `Logic::getName()`.

Reply:
`logicNames` is now a static vector of `std::string`s, so if you return `logicNames[static_cast<int>(aruthType)]` that gives you a `std::string const &` and it is safe, no?

Reply:
Yes.  But `Logic::getName()` returns "QF_UF", and I'm not sure where to store it as a `std::string` if I don't give a copy.

Reply:
True, I see the problem now. Let's postpone it to the next PR.